### PR TITLE
Enforce osusergo build tag for releases

### LINKED ISSUE #6841: Debian 9.6 AMD64: Segfault on start at os/user.Current (1.7.0+)
- Gitea version (or commit ref): 1.6.4
- Git version: 2.11.0
- Operating system: Debian 9.6
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL (MariaDB)
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

After upgrading from gitea version 1.6.4 to 1.7.0 (replacing gitea binary with new one) I'm unable to start gitea. The gitea service fails with the following message (`service gitea status`):
```
● gitea.service - Gitea (Git with a cup of tea)
   Loaded: loaded (/etc/systemd/system/gitea.service; enabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: exit-code) since Fri 2019-05-03 12:27:24 CEST; 1s ago
  Process: 31656 ExecStart=/usr/local/bin/gitea web -c /etc/gitea/app.ini (code=exited, status=2)
 Main PID: 31656 (code=exited, status=2)

```

If I try to run it directly via `gitea web` I get the following:
[Gist](https://gist.github.com/hendrikbl/07b0ad66f30167ab34ceecb4c3b89695
)

More logs:
```
Mai 03 12:18:43 gitea systemd[1]: Started Gitea (Git with a cup of tea).
Mai 03 12:18:43 gitea gitea[29491]: fatal error: unexpected signal during runtime execution
Mai 03 12:18:43 gitea gitea[29491]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x0]
Mai 03 12:18:43 gitea gitea[29491]: runtime stack:
Mai 03 12:18:43 gitea gitea[29491]: runtime.throw(0x19c3d99, 0x2a)
Mai 03 12:18:43 gitea systemd[1]: gitea.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Mai 03 12:18:43 gitea systemd[1]: gitea.service: Unit entered failed state.
Mai 03 12:18:43 gitea systemd[1]: gitea.service: Failed with result 'exit-code'.
```

I get this error with every version greater than 1.6.4. All permissions were set the same with every version I tried. I haven't tried building gitea myself since this is something I only want to do if there really is no other way.

### PR DESCRIPTION
**Pull request by sapk**
Similar to #1690, we need a new tag to use the go version and not the linked cgo version that fail on some distrib. In this case, we are using it to finding the runnning user.

Should fix #6841 and also reported bug on [discourse](https://discourse.gitea.io/t/solved-segfault-due-to-misconfigured-unix-user/1104)

### CODE REVIEW

**Comment by zeripath** [comment]
@sapk are you able to reproduce this problem? We need to ensure that the issue is solved by this addition.

**Comment by sapk** [comment]
@zeripath I can't reproduce the bug exactly since I don't know (yet) which glib is incompatible with which glib version but the output of the linker go in this direction :
```
#Without osuserog
root@a4d05b9c2e6c:/gitea# CGO_ENABLED=1 go build -tags 'netgo sqlite sqlite_unlock_notify' -ldflags '-linkmode external -extldflags "-static"' .
# code.gitea.io/gitea
/tmp/go-link-710302265/000011.o: In function `unixDlOpen':
/root/go/pkg/mod/github.com/mattn/go-sqlite3@v1.10.0/sqlite3-binding.c:38461: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/tmp/go-link-710302265/000015.o: In function `mygetgrouplist':
/workdir/go/src/os/user/getgrouplist_unix.go:16: warning: Using 'getgrouplist' in statically linked applications requires at runtime the shared libraries from theglibc version used for linking
/tmp/go-link-710302265/000014.o: In function `mygetgrgid_r':
/workdir/go/src/os/user/cgo_lookup_unix.go:38: warning: Using 'getgrgid_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/tmp/go-link-710302265/000014.o: In function `mygetgrnam_r':
/workdir/go/src/os/user/cgo_lookup_unix.go:43: warning: Using 'getgrnam_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/tmp/go-link-710302265/000014.o: In function `mygetpwnam_r':
/workdir/go/src/os/user/cgo_lookup_unix.go:33: warning: Using 'getpwnam_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/tmp/go-link-710302265/000014.o: In function `mygetpwuid_r':
/workdir/go/src/os/user/cgo_lookup_unix.go:28: warning: Using 'getpwuid_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
#With osusergo
root@a4d05b9c2e6c:/gitea# CGO_ENABLED=1 go build -tags 'netgo osusergo sqlite sqlite_unlock_notify' -ldflags '-linkmode external -extldflags "-static"' .
# code.gitea.io/gitea
/tmp/go-link-514533019/000011.o: In function `unixDlOpen':
/root/go/pkg/mod/github.com/mattn/go-sqlite3@v1.10.0/sqlite3-binding.c:38461: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
```

The released gitea 1.8 binary seems to work on the docker image debian:9.6. I will try to find a tag that doesn't work.

**Comment by zeripath** [comment]
I've been trying to find the broken version too... I think it's libc6 2.26 but I don't know which version of debian has that.

**Comment by zeripath** [comment]
hmm... Is it possible that this might fix the static build problems for Arm7 might be fixed by osusergo too?

**Comment by sapk** [comment]
On the docker image 9.6  it is libc6 2.24

**Comment by zeripath** [comment]
ubuntu:bionic has 2.27-3ubuntu1 and works

**Comment by sapk** [comment]
For testing, i use :
```
$ cat test.sh
#!/bin/sh
apt-get update
apt-get upgrade -y
apt-get install -y git
dpkg -l libc6
/gitea
wget https://github.com/go-gitea/gitea/releases/download/v1.8.0/gitea-1.8.0-linux-amd64 && chmod +x ./gitea-1.8.0-linux-amd64
docker run -ti --rm -v "$(pwd)/gitea-1.8.0-linux-amd64:/gitea" -v "$(pwd)/test.sh:/test.sh" debian:9.6 /test.sh
```

**Comment by zeripath** [comment]
17.10 works even though its version is 2.26-0ubuntu2.1

File: Makefile [hunk 1]
```diff
@@ -325,7 +325,7 @@ release-windows:
 	@hash xgo > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
 		$(GO) get -u src.techknowlogick.com/xgo; \
 	fi
-	xgo -dest $(DIST)/binaries -tags 'netgo $(TAGS)' -ldflags '-linkmode external -extldflags "-static" $(LDFLAGS)' -targets 'windows/*' -out gitea-$(VERSION) .
+	xgo -dest $(DIST)/binaries -tags 'netgo osusergo $(TAGS)' -ldflags '-linkmode external -extldflags "-static" $(LDFLAGS)' -targets 'windows/*' -out gitea-$(VERSION) .
```
Comment (zeripath) [inline Makefile:328]:
```suggestion
	xgo -dest $(DIST)/binaries -tags 'netgo osusergo static_build $(TAGS)' -ldflags '-linkmode external -extldflags "-static" $(LDFLAGS)' -targets 'windows/*' -out gitea-$(VERSION) .
```

File: Makefile [hunk 2]
```diff
@@ -335,7 +335,7 @@ release-linux:
 	@hash xgo > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
 		$(GO) get -u src.techknowlogick.com/xgo; \
 	fi
-	xgo -dest $(DIST)/binaries -tags 'netgo $(TAGS)' -ldflags '-linkmode external -extldflags "-static" $(LDFLAGS)' -targets 'linux/*' -out gitea-$(VERSION) .
+	xgo -dest $(DIST)/binaries -tags 'netgo osusergo $(TAGS)' -ldflags '-linkmode external -extldflags "-static" $(LDFLAGS)' -targets 'linux/*' -out gitea-$(VERSION) .
```
Comment (zeripath) [inline Makefile:338]:
```suggestion
	xgo -dest $(DIST)/binaries -tags 'netgo osusergo static_build $(TAGS)' -ldflags '-linkmode external -extldflags "-static" $(LDFLAGS)' -targets 'linux/*' -out gitea-$(VERSION) .
```

File: Makefile [hunk 3]
```diff
@@ -345,7 +345,7 @@ release-darwin:
 	@hash xgo > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
 		$(GO) get -u src.techknowlogick.com/xgo; \
 	fi
-	xgo -dest $(DIST)/binaries -tags 'netgo $(TAGS)' -ldflags '$(LDFLAGS)' -targets 'darwin/*' -out gitea-$(VERSION) .
+	xgo -dest $(DIST)/binaries -tags 'netgo osusergo $(TAGS)' -ldflags '$(LDFLAGS)' -targets 'darwin/*' -out gitea-$(VERSION) .
```
Comment (zeripath) [inline Makefile:348]:
```suggestion
	xgo -dest $(DIST)/binaries -tags 'netgo osusergo static_build $(TAGS)' -ldflags '$(LDFLAGS)' -targets 'darwin/*' -out gitea-$(VERSION) .
```

**Comment by zeripath** [comment]
so `static_build` appears to remove a few more warnings too - leaving only `dlopen` warnings which appears to be because sqlite3 can load shared objects so I think that might be irremovable.

**Comment by sapk** [comment]
@zeripath what warning do you got that where remove with static_build ? unixDlOpen was the only I still got using only netgo and osusergo.

**Comment by zeripath** [comment]
oh sorry I'm talking nonsense. I misread your message earlier.

**Comment by zeripath** [comment]
@sapk why don't you come to discord and we can chat a bit more without spamming this

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
I've decided to approve even though we can't prove that this will solve the problem.

Unfortunately it looks like this doesn't solve the arm7 problem (at least on pi2)

**Review by lafriks — APPROVED** [review]

**Comment by zeripath** [comment]
So this will need to be backported to 1.8 in order for us to fix the issue.

**Comment by sapk** [comment]
@zeripath done

**Pull request merged**